sympy/physics/control/routh_table.py

```diff
@@ -0,0 +1,189 @@
+from sympy.matrices.dense import MutableDenseMatrix
+from sympy import Poly
+from sympy import simplify
+from sympy import symbols, Symbol
+
+__all__ = ['RouthHurwitz']
+
+
+class RouthHurwitz(MutableDenseMatrix):
+    r"""
+        A class for creating a Routh-Hurwitz table from a given polynomial.
+        It handle special cases with methods discussed in [1].
+
+        Note: When a row of the table is zero, the property ``zero_rows_case`` is set to True.
+
+        Explanation
+        ============
+
+        TODO: Explain the special cases and why the ``zero_rows_case`` property is important.
+
+        Parameters
+        ==========
+
+        polynomial : :py:class:`~.Expr`, :py:class:`~.Number`
+            The polynomial whose Routh-Hurwitz table is to be created.
+        var : :py:class:`~.Symbol`
+            The symbol representing the variable in the polynomial.
+        infinitesimal_element : None, :py:class:`~.Symbol`, optional
+            The symbol representing the infinitesimal element for the first column zero case.
+            If not provided, a default symbol ``epsilon`` will be used.
+
+        Examples
+        ========
+
+        >>> from sympy import symbols
+        >>> from sympy.physics.control import RouthHurwitz
+        >>> b1, b2, b3, b4 = symbols('b_1 b_2 b_3 b_4')
+        >>> s = symbols("s")
+
+        Here is a generic example of how to use the ``RouthHurwitz`` class:
+
+        >>> p = b1*s**3 + b2*s**2 + b3*s + b4
+        >>> RouthHurwitz(p, s)
+        Matrix([
+        [               b_1, b_3],
+        [               b_2, b_4],
+        [-b_1*b_4/b_2 + b_3,   0],
+        [               b_4,   0]])
+        >>> RouthHurwitz(p, s).first_column
+        Matrix([
+        [               b_1],
+        [               b_2],
+        [-b_1*b_4/b_2 + b_3],
+        [               b_4]])
+
+        Here you can see how the table appears in the first column zero case:
+
+        >>> p1 = s**4 + s**3 + 3*s**2 + 3*s + 3
+        >>> RouthHurwitz(p1, s)
+        Matrix([
+        [            1, 3, 3],
+        [            1, 3, 0],
+        [      epsilon, 3, 0],
+        [3 - 3/epsilon, 0, 0],
+        [            3, 0, 0]])
+        >>> RouthHurwitz(p1, s).zero_rows_case
+        False
+
+        Here you can see how the table appears in the full row zero case (poles with only imaginary part):
+
+        >>> p2 = s**6 + 2*s**5 + 8*s**4 + 12*s**3 + 20*s**2 + 16*s + 16
+        >>> RouthHurwitz(p2, s)
+        Matrix([
+        [  1,  8, 20, 16],
+        [  2, 12, 16,  0],
+        [  2, 12, 16,  0],
+        [  8, 24,  0,  0],
+        [  6, 16,  0,  0],
+        [8/3,  0,  0,  0],
+        [ 16,  0,  0,  0]])
+        >>> RouthHurwitz(p2, s).zero_rows_case
```

Comment:
I don't understand what "zeros_rows_case" means and why it would return a boolean.